### PR TITLE
RDART-980: Add tests for collections in `Mixed` with synced realms

### PR DESCRIPTION
**Pull request by elle-j**
Update the collections in `Mixed` tests with tests using synced realms.

### CODE REVIEW

**Review by nielsenko — APPROVED** [review]
LGTM

There is a lot of almost copy-paste blocks of expects. You could consider finding a way to abstract over these.

You could also consider adding some tests for conflict resolution, by

- pausing both realms,
- do conflicting edits,
- resume,
- verify outcome.

I know there are some corner cases around clearing collections and changing types at the same time.

File: packages/realm_dart/test/realm_value_test.dart [hunk 1]
```diff
@@ -15,30 +15,41 @@ class _TuckedIn {
   int x = 42;
 }
 
-@RealmModel()
```
[the remark this answers is not in the document]
Comment (elle-j) [reply packages/realm_dart/test/realm_value_test.dart:18]: Haha, the fun is over 😉

File: packages/realm_dart/test/realm_value_test.dart [hunk 2]
```diff
@@ -15,30 +15,41 @@ class _TuckedIn {
...
+      mutableSubscriptions.add(realm.query<ObjectWithInt>(r'differentiator = $0', [differentiator]));
+    });
+    await realm.subscriptions.waitForSynchronization();
+
+    return realm;
+  }
+
+  Future<(Realm, Realm)> logInAndGetSyncedRealms(AppConfiguration appConfig, ObjectId differentiator) async {
+    final realm1 = await logInAndGetSyncedRealm(appConfig, differentiator);
+    final realm2 = await logInAndGetSyncedRealm(appConfig, differentiator);
```
[the remark this answers is not in the document]
Comment (elle-j) [reply packages/realm_dart/test/realm_value_test.dart:39]: Yes it utilizes the preexisting helper function for registering and logging in a new user.

File: packages/realm_dart/test/realm_value_test.dart [hunk 3]
```diff
@@ -227,76 +287,114 @@ void main() {
...
-      Stuff(),
-      now,
-      ObjectId.fromTimestamp(now),
-      Uuid.v4(),
-      Decimal128.fromInt(128),
-    ];
-    final realm = getMixedRealm();
-    final realmValues = values.map(RealmValue.from);
-    realm.write(() => realm.add(AnythingGoes(manyAny: realmValues, oneAny: realmValues.last)));
+    baasTest('[BaaS] Roundtrip', (appConfig) async {
```
[the remark this answers is not in the document]
Comment (elle-j) [reply packages/realm_dart/test/realm_value_test.dart:316]: Good point!

File: packages/realm_dart/test/realm_value_test.dart [hunk 4]
```diff
@@ -80,42 +109,72 @@ void main() {
     }
 
     test('Roundtrip object', () {
-      final stuff = Stuff(i: 123);
+      final stuff = ObjectWithInt(ObjectId(), i: 123);
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply packages/realm_dart/test/realm_value_test.dart:112]: No.. I was talking about the variable still being called `stuff`.

[the remark this answers is not in the document]
Comment (elle-j) [reply packages/realm_dart/test/realm_value_test.dart:112]: I can update those 👍

**Review by nielsenko — APPROVED** [review]
Even better with the new tests
